### ISSUE TITLE
annotate --finetune produces an empty annotation on Paramecium bursaria

### ISSUE THREAD

**Issue by guyleonard** [Gaius-Augustus/Vipsania#5]
## Summary

`vipsania annotate --finetune` completes without error on this genome and writes
an empty GFF3 (16 bytes, header only) with 0-byte protein and CDS files. The same
model on the same genome without `--finetune` predicts 3,212 genes.

## Versions

- Vipsania: `translation_table` branch, commit `1aaae1f`
- bricks2marble: `translation_table` branch, commit `d2938de`
- TensorFlow 2.19.1, Python 3.12, 4x NVIDIA A40

## Genome

*Paramecium bursaria* 186b, GCA_053455385.1. 328 sequences, 29.9 Mb total,
4.14% soft-masked (lowercase; no N).

## Without finetuning: genes predicted

Whole genome:

```bash
vipsania annotate Alveolata pb186b.scaffolds.fa \
  -o control.gff3 --protein control.aa.fa --coding control.cds.fa
```

3,212 proteins, mean length 70 aa.

Single scaffold:

```bash
vipsania annotate Alveolata pb186b.scaffolds.fa -i pb186b_2025_1 \
  -o probe.gff3 --coding probe.cds.fa --translation_table 6
```

39 gene, 39 mRNA, 55 CDS, 55 exon.

The reference annotation for this genome has 21,279 genes, so the
non-finetuned prediction is poor. Probably expected?

## With finetuning: empty

```bash
vipsania annotate Alveolata pb186b.scaffolds.fa --finetune --finetune_epochs 1 \
  -o probe.gff3 --coding probe.cds.fa
```

Annotation runs through every stage for all three sequence groups, no error:

```
[0.0748s] Start initial prediction of 301 chunks.
[402.4790s] Searching for errors.
[402.5325s] Forming transcripts.
[402.9509s] Creating annotation.
[402.9533s] Calling postprocessing function.
[402.9540s] Writing annotation to file.
[402.9542s] Done.
```

Output: 16-byte GFF3, 0-byte CDS and protein files.

## Variants tried

All finetuned runs below give the same empty output.

| run | epochs | `--translation_table` | masking |
|---|---|---|---|
| `annotate --finetune` | 1 | 6 | unmasked |
| `annotate --finetune` | 1 | default | soft-masked |
| `annotate --finetune --relax_repeats` | 10 | default | soft-masked |
| `annotate --finetune --relax_repeats` | 10 | 6 | soft-masked |

So I don't think it is specific to `--translation_table`, or specific to epoch count, and
probably not caused by the soft-masking.

For the 10-epoch, table-6 run the config written to `finetuning_1/config.json`
was correct:

```python
{'translation_table': 6, 'start_codons': [['ATG', 1.0]], 'stop_codons': [['TGA', 1.0]]}
```

Re-annotating the single test scaffold directly from that checkpoint also gave no
features?

```bash
vipsania annotate finetuning_1 --model_dir <dir> pb186b.scaffolds.fa \
  -i pb186b_2025_1 -o ft_probe.gff3 --coding ft_probe.cds.fa
```

I also ran `train --resume --fork` on this genome for 10 epochs, setting the stop
codons through a training config rather than the CLI flag, and annotation from
that model were also empty.

## Questions

- Is there a way to distinguish "the model predicted no genes" from "all
  predictions were removed by postprocessing"? The log prints the same stages in
  both cases, which made this hard to narrow down.
- At 29.9 Mb this is a small genome, it is also AT rich (26% GC). Is there a lower bound on genome size for
  finetuning to be useful, and is undertraining the likely explanation here? Even
  so, I would have expected a finetuned model to predict something rather than
  nothing.

Happy to share the assembly if a test case is useful (or NCBI:GCA_053455385.1).
